newrelic/resource_newrelic_workflow.go

```diff
@@ -0,0 +1,371 @@
+package newrelic
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/newrelic/newrelic-client-go/pkg/workflows"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/newrelic/newrelic-client-go/pkg/ai"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+)
+
+func resourceNewRelicWorkflow() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNewRelicWorkflowCreate,
+		ReadContext:   resourceNewRelicWorkflowRead,
+		UpdateContext: resourceNewRelicWorkflowUpdate,
+		DeleteContext: resourceNewRelicWorkflowDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "(Required) The name of the workflow.",
+			},
+			"destination_configurations": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Workflow's destination configuration.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// Required
+						"channel_id": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "(Required) Destination's channel id.",
+						},
+
+						// Computed
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "(Required) Destination's name.",
+						},
+						"type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: fmt.Sprintf("(Required) The type of the destination. One of: (%s).", strings.Join(listValidWorkflowsDestinationTypes(), ", ")),
+						},
+					},
+				},
+			},
+			"issues_filter": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "Workflow's destination configuration.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// Required
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "(Required) Filter's name.",
+						},
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice(listValidWorkflowsFilterTypes(), false),
+							Description:  fmt.Sprintf("(Required) The type of the filter. One of: (%s).", strings.Join(listValidWorkflowsFilterTypes(), ", ")),
+						},
+
+						// Optional
+						"predicates": {
```

Comment:
another case where this is a block and so should be using singular form `predicate` since it's a block that defines a single predicate configuration.

Reply:
should we also do it to notification channels and destinations `properties`?

Reply:
fixed in this pr: https://github.com/newrelic/terraform-provider-newrelic/pull/1990